jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -1351,31 +1318,6 @@ ecma_is_string_magic_longpath (const ecma_string_t *string_p, /**< ecma-string *
 } /* ecma_is_string_magic_longpath */
 
 /**
- * Long path part of ecma_is_ex_string_magic
- *
- * Converts passed ecma-string to zt-string and
- * checks if it is equal to one of magic string
- *
- * @return true - if magic string equal to passed string was found,
- *         false - otherwise.
- */
-static bool
-ecma_is_ex_string_magic_longpath (const ecma_string_t *string_p, /**< ecma-string */
-                                  lit_magic_string_ex_id_t *out_id_p) /**< [out] external magic string's id */
```

Comment:
Yeah, that's a tricky one... The catch is that these two functions have almost no effect on external magic string support. If you check their "non-external" counterpart, `ecma_is_string_magic`, you can find out that it's used at a single place only: in `TRY_TO_INSTANTIATE_PROPERTY_ROUTINE_NAME` (in ecma-builtin-internal-routines-template.inc.h). There we try to find a magic string id associated with a builtin. All that is basically hard-wired/built into jerry. And there is no other use for `ecma_is_string_magic`. And I cannot see any valid scenario where external magic strings could be used in a similar fashion.

The real handling of magic strings is in `lit/`. The truly useful functions, like `lit_is_utf8_string_magic` or `it_is_ex_utf8_string_magic` is not touched by this patch.